url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/TestForHttpResponse.java

```diff
@@ -0,0 +1,90 @@
+package net.hetimatan.util.http;
+
+
+import java.io.IOException;
+
+import net.hetimatan.io.file.MarkableFileReader;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.util.http.HttpRequestLine;
+import net.hetimatan.util.http.HttpResponse;
+import junit.framework.TestCase;
+
+public class TestForHttpResponse extends TestCase {
+
+	public void testEncode001_emptyBody() throws IOException {
+		String httpVersion = HttpRequestLine.HTTP11;
+		String statusCode = "200";
+		String reasonPhrase = "OK";
+		HttpResponse response = new HttpResponse(httpVersion, statusCode,
+				reasonPhrase);
+		CashKyoroFile output = new CashKyoroFile(512);
+		response.encode(output.getLastOutput());
+		output.seek(0);
+		byte[] buffer = new byte[(int) output.length()];
+		int len = output.read(buffer);
+		String tag = new String(buffer, 0, len);
+		assertEquals("HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n", tag);
+	}
+
+	public void testEncode002_Body() throws IOException {
+		String httpVersion = HttpRequestLine.HTTP11;
+		String statusCode = "200";
+		String reasonPhrase = "OK";
+		HttpResponse response = new HttpResponse(httpVersion, statusCode,
+				reasonPhrase);
+		String content = "hello world\r\n";
+		response.setContent(content.getBytes());
+		CashKyoroFile output = new CashKyoroFile(512);
+		response.encode(output.getLastOutput());
+		output.seek(0);
+		byte[] buffer = new byte[(int) output.length()];
+		int len = output.read(buffer);
+		String tag = new String(buffer, 0, len);
+		assertEquals("HTTP/1.1 200 OK\r\nContent-Length: "
+				+ content.getBytes().length + "\r\n\r\nhello world\r\n", tag);
+	}
+
+	public void testDecode001() throws IOException {
+		CashKyoroFile base = new CashKyoroFile(512);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *base*. Currently, there are execution paths that do not contain closure statements, for example, when CashKyoroFile.addChunk throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).